pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/TimeSeriesIntegrationTest.java

```diff
@@ -0,0 +1,306 @@
+
+  @Override
+  public String getTableName() {
+    return DEFAULT_TABLE_NAME;
+  }
+
+  @Override
+  public long getCountStarResult() {
+    return 1000L;
+  }
+
+  @Override
+  public Schema createSchema() {
+    return new Schema.SchemaBuilder().setSchemaName(getTableName())
+      .addSingleValueDimension(TS_COLUMN, FieldSpec.DataType.LONG)
+      .addSingleValueDimension(CITY_COLUMN, FieldSpec.DataType.LONG)
+      .addSingleValueDimension(PLATFORM_COLUMN, FieldSpec.DataType.STRING)
+      .addSingleValueDimension(REFERRAL_COLUMN, FieldSpec.DataType.BOOLEAN)
+      .addSingleValueDimension(VIEWS_COLUMN, FieldSpec.DataType.LONG)
+      .build();
+  }
+
+  private org.apache.avro.Schema.Field createAvroField(String name, org.apache.avro.Schema.Type type) {
+    return new org.apache.avro.Schema.Field(name, org.apache.avro.Schema.create(type), null, null);
+  }
+
+  public File createAvroFile()
+    throws Exception {
+    org.apache.avro.Schema avroSchema = org.apache.avro.Schema.createRecord("myRecord", null, null, false);
+    avroSchema.setFields(ImmutableList.of(
+      createAvroField(TS_COLUMN, org.apache.avro.Schema.Type.LONG),
+      createAvroField(CITY_COLUMN, org.apache.avro.Schema.Type.LONG),
+      createAvroField(PLATFORM_COLUMN, org.apache.avro.Schema.Type.STRING),
+      createAvroField(REFERRAL_COLUMN, org.apache.avro.Schema.Type.BOOLEAN),
+      createAvroField(VIEWS_COLUMN, org.apache.avro.Schema.Type.LONG)
+    ));
+
+    File avroFile = new File(_tempDir, "data.avro");
+    try (DataFileWriter<GenericData.Record> writer = new DataFileWriter<>(new GenericDatumWriter<>(avroSchema))) {
+      writer.create(avroSchema, avroFile);
+      for (int i = 0; i < getCountStarResult(); i++) {
+        writer.append(getRecord(avroSchema, i));
+      }
+    }
+    return avroFile;
+  }
+
+  private static GenericData.@NotNull Record getRecord(org.apache.avro.Schema avroSchema, int i) {
+    GenericData.Record record = new GenericData.Record(avroSchema);
+    // Do not set DATA_START_TIME_SEC for easier assertion of values.
+    record.put(TS_COLUMN, (DATA_START_TIME_SEC + 1 + i) * 1000L);
+    record.put(CITY_COLUMN, i % 5);
+    record.put(PLATFORM_COLUMN, PLATFORMS[i % PLATFORMS.length]);
+    record.put(REFERRAL_COLUMN, (i % 2) == 0);
+    // Alternate between VIEWS_MIN_VALUE and VIEWS_MAX_VALUE.
+    record.put(VIEWS_COLUMN, VIEWS_MIN_VALUE + (VIEWS_MAX_VALUE - VIEWS_MIN_VALUE) * (i % 2));
+    return record;
+  }
+
+  @BeforeClass
+  public void setUp()
+    throws Exception {
+    LOGGER.warn("Setting up integration test class: {}", getClass().getSimpleName());
+    TestUtils.ensureDirectoriesExistAndEmpty(_tempDir, _segmentDir, _tarDir);
+
+    // Start the Pinot cluster
+    startZk();
+    startController();
+    startBroker();
+    startServer();
+
+    if (_controllerRequestURLBuilder == null) {
+      _controllerRequestURLBuilder =
+        ControllerRequestURLBuilder.baseUrl("http://localhost:" + getControllerPort());
```

Comment:
nit: format. I think the argument will be on a new line and the preceding tokens can stay on the same line.

Idk if we have a clear definition of Pinot Style.. we really should define and enforce it at some point